zeebe/stream-platform/src/test/java/io/camunda/zeebe/stream/impl/StreamProcessorTest.java

```diff
@@ -407,10 +409,18 @@ public void shouldProcessFollowUpEventsAndCommands() {
     assertThat(firstRecord.getSourceEventPosition()).isEqualTo(-1);
     final var firstRecordPosition = firstRecord.getPosition();
 
+    final var firstRecordMetadata = new RecordMetadata();
+    firstRecord.readMetadata(firstRecordMetadata);
+    assertThat(firstRecordMetadata.getRequestId()).isEqualTo(requestId);
+
     await("should write follow up events")
         .untilAsserted(() -> assertThat(logStreamReader.hasNext()).isTrue());
     while (logStreamReader.hasNext()) {
-      assertThat(logStreamReader.next().getSourceEventPosition()).isEqualTo(firstRecordPosition);
+      final LoggedEvent followup = logStreamReader.next();
+      assertThat(followup.getSourceEventPosition()).isEqualTo(firstRecordPosition);
```

Comment:
👍 Nice one. This exactly verifies if we are able to propagate the request id to follow up events. It mocks the record processor but not the processing result builder where the logic lies.